library/math/conversions.lisp

```diff
@@ -0,0 +1,566 @@
+      (when (> x (into (the I32 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x (into (the I32 minBound)))
+        (return (Err "value out of range")))
+
+      (Ok (lisp I32 (x) x))))
+
+  (define-instance (TryInto I64 U64)
+    (define (tryInto x)
+      (when (< x 0)
+        (return (Err "value out of range")))
+
+      (Ok (lisp U64 (x) x))))
+
+  (define-instance (Into I64 Integer)
+    (define (into x)
+      (lisp Integer (x) x)))
+
+  ;;
+  ;; Conversions from Integer
+  ;;
+
+  (define-instance (TryInto Integer U8)
+    (define (tryInto x)
+      (when (> x (into (the U8 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x 0)
+        (return (Err "value out of range")))
+
+      (Ok (lisp U8 (x) x))))
+
+  (define-instance (TryInto Integer I8)
+    (define (tryInto x)
+      (when (> x (into (the I8 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x (into (the I8 minBound)))
+        (return (Err "value out of range")))
+
+      (Ok (lisp I8 (x) x))))
+
+  (define-instance (TryInto Integer U16)
+    (define (tryInto x)
+      (when (> x (into (the U16 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x 0)
+        (return (Err "value out of range")))
+
+      (Ok (lisp U16 (x) x))))
+
+  (define-instance (TryInto Integer I16)
+    (define (tryInto x)
+      (when (> x (into (the I16 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x (into (the I16 minBound)))
+        (return (Err "value out of range")))
+
+      (Ok (lisp I16 (x) x))))
+
+  (define-instance (TryInto Integer U32)
+    (define (tryInto x)
+      (when (> x (into (the U32 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x 0)
+        (return (Err "value out of range")))
+
+      (Ok (lisp U32 (x) x))))
+
+  (define-instance (TryInto Integer I32)
+    (define (tryInto x)
+      (when (> x (into (the I32 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x (into (the I32 minBound)))
+        (return (Err "value out of range")))
+
+      (Ok (lisp I32 (x) x))))
+
+  (define-instance (TryInto Integer U64)
+    (define (tryInto x)
+      (when (> x (into (the U64 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x 0)
+        (return (Err "value out of range")))
+
+      (Ok (lisp U64 (x) x))))
+
+  (define-instance (TryInto Integer I64)
+    (define (tryInto x)
+      (when (> x (into (the I64 maxBound)))
+        (return (Err "value out of range")))
+
+      (when (< x (into (the I64 minBound)))
+        (return (Err "value out of range")))
+
+      (Ok (lisp I64 (x) x)))))
+
+(coalton-toplevel
+  (define-instance (Into IFix Integer)
+    (define (into x)
+      (lisp Integer (x) x)))
+
+  (define-instance (Into UFix Integer)
+    (define (into x)
+      (lisp Integer (x) x))))
+
+(coalton-toplevel
+  (declare integer->double-float (Integer -> Double-Float))
```

Comment:
This can result in an overflow for large integers that can't be represented by floats. Which is fine, but it should probably be documented, as well as perhaps a use-case to justify it. Otherwise, `fromInt` has achieves the same thing but with infs instead of errors. Although, `Into` instances of fixed-length integers would be completely safe.

Reply:
Yeah we should just use fron int.